design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,297 @@
+	DestinationPvc string `json:"destinationPvc,omitempty" valid:"required"`
+	// ReclaimPolicySourcePvc describes how the source volumes will be
+	// treated after a successful migration
+	ReclaimPolicySourcePvc ReclaimPolicySourcePvc `json:"reclaimPolicySourcePvc,omitempty"`
+}
+
+type VolumeMigrationSpec struct {
+	// MigratedVolumes is a list of volumes to be migrated
+	MigratedVolume []MigratedVolume `json:"migratedVolume,omitempty"`
+}
+
+type VolumeMigrationState struct {
+	// MigratedVolumes is a list of volumes to be migrated
+	MigratedVolume []MigratedVolume `json:"migratedVolume,omitempty"`
+	// VirtualMachineMigrationName state of the virtual machine migration
+	// triggered by the storage migration
+	VirtualMachineMigrationName string `json:"virtualMachineMigrationName,omitempty"`
+	// VirtualMachineMigrationState state of the virtual machine migration
+	// triggered by the storage migration
+	VirtualMachineInstanceName string `json:"virtualMachineInstanceName,omitempty"`
+	Succeeded bool `json:"succeeded,omitempty"`
+	Failed bool `json:"failed,omitempty"`
+	// The time the migration action began
+	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
+	// The time the migration action ended
+	EndTimestamp *metav1.Time `json:"endTimestamp,omitempty"`
+}
+
+type RejectedMigratedVolume struct {
+	SourcePvc string `json:"sourcePvc,omitempty" valid:"required"`
+	Reason    string `json:"reason,omitempty" valid:"required"`
+}
+
+type VolumeMigrationStatus struct {
+    // VolumeMigrationStates describes the migration of the migrated volumes
+	VolumeMigrationStates []VolumeMigrationState   `json:"volumeMigrationStates,omitempty"`
+	// PendingVolumes are the migrated volumes that aren't associated to a VMI and cannot by migrated with this APIs
+	PendingVolumes        []MigratedVolume         `json:"pendingVolumes,omitempty"`
+	// RejectedVolumes are a list of source migrated volumes that cannot be migrated
+	RejectedVolumes       []RejectedMigratedVolume `json:"rejectedVolumes,omitempty"`
+	Total                 int                      `json:"total,omitempty"`
+	Failed                int                      `json:"failed,omitempty"`
+	Succeeded             int                      `json:"succeeded,omitempty"`
+	NotStarted            int                      `json:"notStarted,omitempty"`
+	Running               int                      `json:"running,omitempty"`
+	Pending               int                      `json:"pending,omitempty"`
+	Rejected              int                      `json:"rejected,omitempty"`
+}
+```
+
+
+The `VolumeMigration` expects a list of source and destination PVCs and the destination PVCs needs to be created either manually by the user or other tools.
+
+According to the 'ReclaimPolicySourcePvc', KubeVirt will determine how to handle the source PVCs. For the `Delete` policy, the source PVC will be automatically deleted while for the `Retain` policy, the PVC will be left untouched.
```

Comment:
What about if the policy is unset? the struct allows for an empty value.

Reply:
We can either set a default or validate it in the webhook. I personally prefer the latter.